File: 03_calculate_big_o.js
// This funChallenge has BIG O(3 + 4n)

function anotherFunction() {
  console.log('in another function')
}

function funChallenge(input) {
  let temp = 10 //O(1)
  temp = 50 + 3 //O(1)

  //O(n) - for the for loop itself
  for (let i = 0; i < input.length; i++) {
    anotherFunction() //O(n)
    let stranger = true //O(n)
    a++ //O(n)
  }
  return a //O(1)
}

funChallenge('hello') //BIG O(3 + 4n)

File: 03_calculate_big_o.py
# This funChallenge has BIG O(4 + 4n)
# ***commented out print - used only for debugging purpose

def anotherFunction():
  print('in another function')

def funchallenge(input):
    temp = 10 #O(1)
    temp = temp +50 #O(1)
    a = 0 #O(1)

    for i in range(len(input)): #O(n)
        anotherFunction() #O(n) because every loop on n times this function call 
        tempAnother = True #O(n)
        a += 1 #O(n)
    return a #O(1)

funchallenge('hello') # BIG O (4 + 4n)
# print(funchallenge('hello'))
